### DISCUSSION TITLE
v3 settings can't be imported into v4

### DISCUSSION THREAD

**Discussion by wattux** [bluecaret/carettab#343] [Help and Support] [answered]
Got v4 today. When I attempt to import the settings from v3, I get the following message (via console):

```
index.html-513a145c.js:335 Uncaught (in promise) ReferenceError: $t is not defined
    at g.onload (index.html-513a145c.js:335:59681)
```

It's this code part:

```
if (confirm($t("advanced.importOldVersionWarning"))) {
    e.isLoading = !1,
    bo(u);
    return
}
```

I'd really like to not start all over again :)

Here's my settings file. I stripped only the bookmark links. (appended a ".txt" suffix since Github disallows ".json" -.-)

[carettab-settings-example.json.txt](https://github.com/bluecaret/carettab/files/12181868/carettab-settings-example.json.txt)

**Comment by bluecaret** [comment, marked as answer]
I'm very sorry for this. First there should have been a message saying that the settings config is not compatible, as v3 settings are very different from v4. However, the update should have tried migrating as much as possible and add them as disabled widgets in the dashboard for things like quick links and quotes. If this did not happen I can take your settings and do that manually for you. Be aware this won't work for a lot of design settings though as widgets are placed and configured very differently now.

**Reply by wattux** [reply]
@bluecaret Thanks for taking the time! You shouldn't waste your time with converting a single persons settings file, so I'll bite the bullet and set things up again. Things do change and nothing is forever :P

**Reply by bluecaret** [reply]
I appreciate that! Hopefully this will not be an issue in the future. Thank you for using CaretTab! Feel free to reach out if you need any other assistance.
